File: c++/Array_unidimencional.cpp
#include<iostream>
using namespace std;

//clase array unidimencional
//ingreso 
class ArrgUni{
	public:
		int arrg[5];		
		void addData(int x){
			arrg[0]=x;
		}
};

//sub clase para el recorrido
class Recorrido:public ArrgUni {
	public:
		void recorridoArray(){
			for(int i=0;i<5;i++){
				cout<<"Recorrido :"<<arrg[i]<<endl;
			}
		}
};

int main(){
	ArrgUni Array;
	Recorrido recorrido;
	Array.addData(1);
	Array.addData(2);
	Array.addData(3);
	recorrido.recorridoArray();
	return 0;
}

File: java/Array_unidimencional.java
//arraya unidimencional

public class Array_unidimencional{
   public static void main(String[] args){
      int array[] = new int[5]; 
      int i=0;
      
      //tamaño del array
      int tamanoarray = arraySize(array);
      System.out.println(tamanoarray);
      
      //insercion de elementos
      while(i<5){
         insercion(i,array);
         i++;
      }      
      //recorriod del array
      recorrido(array);
      
   }
   
   //funcion para obtener el tamaño del array
   static int arraySize(int arrg[]){
      return arrg.length;
   }
   //metodo para insertar elementos en el array
   static void insercion(int x,int arr[]){
      for(int i=0;i<5;i++){
         arr[i]=x;
      }
   }
   //recorrido del arreglo
   static void recorrido(int arr[]){
      for(int i=0; i<=4;i++){
         System.out.println("Elementos en la posicion ["+i+"] "+arr[i]);
      }
   }
}